src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/hyjal_trash.cpp

```diff
@@ -1503,6 +1524,47 @@ class alliance_rifleman : public CreatureScript
 
 };
 
+class alliance_horde_guard : public CreatureScript
+{
+public:
+    alliance_horde_guard() : CreatureScript("alliance_horde_guard") { }
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return GetHyjalAI<alliance_horde_guardAI>(creature);
+    }
+
+    struct alliance_horde_guardAI : public ScriptedAI
+    {
+        alliance_horde_guardAI(Creature* creature) : ScriptedAI(creature) {}
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            Talk(TRASH_SAY_DEATH);
+            me->SetRespawnDelay(me->GetRespawnDelay()+480); // prevent respawn in combat,add wave spawn time to delay respawn
+        }
+
+        void KilledUnit(Unit* /*victim*/) override
+        {
+            Talk(TRASH_SAY_SLAY);
+        }
+
+        void MoveInLineOfSight(Unit* who) override
+        {
+            if (!who || me->GetVictim())
+                return;
+
+            if (me->IsValidAttackTarget(who))
+            {
+                if (me->IsWithinDistInMap(who, 10))
+                    AttackStart(who);
+            }
+        }
```

Comment:
What you're trying to fix by overriding MoveInLineOfSight?

Reply:
Ths guards are not autoattack.They just stand until take damage.A bit defference from retail version.

Reply:
We can't override it each time something doesn't work

Reply:
Can't reproduce with mobs at first base

Reply:
That sounds like a generic bug or something related to specific creature

Reply:
I think I expressed.Guards not autoattack when trashs move in sight,they just stand until taken any damage from trashs.After guards kill the target,they immediately returned to its original position without autoattack another targe in sight.

Reply:
Generic bug or not a bug, it happens everywhere

Reply:
Ok。I remove these code lines.